src/arrays.js

```diff
@@ -3,38 +3,80 @@
 
 
 const each = (elements, cb) => {
+  for (let i = 0; i < elements.length; i++) {
+    cb(elements[i], i);
+  }
   // Iterates over a list of elements, yielding each in turn to the `cb` function.
   // This only needs to work with arrays.
   // based off http://underscorejs.org/#each
 };
 
 const map = (elements, cb) => {
+  const array = [];
+  for (let i = 0; i < elements.length; i++) {
+    array.push(cb(elements[i]));
+  }
+  return array;
   // Produces a new array of values by mapping each value in list through a transformation function (iteratee).
   // Return the new array.
 };
 
 const reduce = (elements, cb, memo) => {
+  let i = 0;
+  if (!memo) {
+    memo = elements[0];
+    i = 1;
+  }
+  let total = memo; // 10
+  for (; i < elements.length; i++) {
+    total += elements[i];
+  }
+  return total;
+};
   // Combine all elements into a single value going from left to right.
   // Elements will be passed one by one into `cb`.
   // `memo` is the starting value.  If `memo` is undefined then make `elements[0]` the initial value.
-};
+
 
 const find = (elements, cb) => {
   // Look through each value in `elements` and pass each element to `cb`.
   // If `cb` returns `true` then return that element.
   // Return `undefined` if no elements pass the truth test.
+  for (let i = 0; i < elements.length; i++) {
+    if (cb(elements[i])) { // tai says look at this
```

Comment:
This works fine.   Because `cb` is returning `true` or `false` it will pass or fail the `if` statement